core/utils/object_pool.hpp

```diff
@@ -118,15 +114,16 @@ class concurrent_stack : private util::noncopyable {
 
       new_head.node = &new_node;
```

Comment:
This line could be moved before the loop.

Reply:
good catch!